### PR TITLE
add support for writing the 'and!' expressions and patterns

### LINKED ISSUE #690: Support and keyword in computation expression
Issue created from [fantomas-ui](https://jindraivanek.gitlab.io/fantomas-ui/#?fantomas=preview&code=PTAECUFMEMBMGdQBcAWlQDdoBsCulEB7AM1AA8AaUAT1GgDtZQAvUAY0PrdwCcfJ6SbNSqoBdAA4TsASwKhSSQsjQBbALAAoCdB45skbKADeW0OdAGkAQnKgAvKHjZCAdygBHfPCQANABQAlGYWDLC2tI7Obp7eSACaQSHmYbasUS7ukF4ESABaSZoWoPxIvPQKdrTMWgC+QA)

Support https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1063-support-letbang-andbang-for-applicative-functors.md

#### Code

```fsharp
// Reads the values of x, y and z concurrently, then applies f to them
parallel {
    let! x = slowRequestX()
    and! y = slowRequestY()
    and! z = slowRequestZ()
    return f x y z
}
```
            

#### Result

```fsharp
// Reads the values of x, y and z concurrently, then applies f to them
``parallel`` {
    let! x = slowRequestX()
    return f x y z }

```
            
#### Options

Fantomas Next - 3.2.0-2/10/2020

| Name | Value |
| ---- | ----- |
| `IndentOnTryWith` | `false` |
| `IndentSpaceNum` | `4` |
| `KeepNewlineAfter` | `false` |
| `MaxIfThenElseShortWidth` | `40` |
| `PageWidth` | `120` |
| `ReorderOpenDeclaration` | `false` |
| `SemicolonAtEndOfLine` | `false` |
| `SpaceAfterComma` | `true` |
| `SpaceAfterSemicolon` | `true` |
| `SpaceAroundDelimiter` | `true` |
| `SpaceBeforeArgument` | `true` |
| `SpaceBeforeColon` | `false` |
| `StrictMode` | `false` |

### PR DESCRIPTION
**Pull request by baronfel**
Fixes #690 by writing the and! expressions in line!

### CODE REVIEW

File: src/Fantomas/CodePrinter.fs [hunk 1]
```diff
@@ -1439,9 +1439,14 @@ and genExpr astContext synExpr =
         addParenIfAutoNln e1 (genExpr astContext) -- sprintf " <- " +> genExpr astContext e2
 
     | LetOrUseBang(isUse, p, e1, ands, e2) ->
+
+        let genAndList astContext (ands: list<SequencePointInfoForBinding * bool * bool * SynPat * SynExpr * range>) =
+            (sepNone, ands)
+            ||> List.fold (fun fn (_,_,_,pat,expr,_) -> fn +> !- "and! " +> genPat astContext pat -- " = " +> genExpr astContext expr +> sepNln)
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs]: I think you can use the `col` helper function (or a variant) instead of the List.fold.

Comment (nojaf) [reply src/Fantomas/CodePrinter.fs]: See https://github.com/fsprojects/fantomas/blob/master/src/Fantomas/Context.fs#L283

Comment (baronfel) [reply src/Fantomas/CodePrinter.fs]: Yep, that worked great! Thanks for the pointer.

**Comment by baronfel** [comment]
Rebased on top of latest master and made the `col` change as requested.

**Pull request merged**